src/components/Button.tsx

```diff
@@ -91,15 +90,11 @@ export function Button(props: ButtonProps) {
     );
 
   // If we're disabled b/c of a non-boolean ReactNode, or the caller specified tooltip text, then show it in a tooltip
-  if ((isDisabled && typeof disabled !== "boolean") || tooltip) {
-    return (
-      <Tooltip title={isDisabled && typeof disabled !== "boolean" ? disabled : tooltip} placement="top">
-        {button}
-      </Tooltip>
-    );
-  }
-
-  return button;
+  return maybeTooltip({
+    title: isDisabled && typeof disabled !== "boolean" ? disabled : tooltip,
```

Comment:
This `isDisabled && typeof ...` always takes me already to parse; could we change `title` to be `title: string | [string | undefined, string | undefined]`, so then this could be:

```
title: [disabled, tooltip]
```

And internally `maybeTooltip` can do the `if title instanceof Array` then pick the first value that is not `undefined`. Or not show the tooltip if both are undefined...

Reply:
Or, if you don't want to push that tuple/array-handling into `maybeTooltip` could make a separate helper method like:

```
title: resolveTitle(disabled, tooltip)
```

Reply:
Or I guess it could be `title: Maybe<string> | Array<Maybe<string>>` (don't have to use a `Maybe` alias but just easier to type :-))

Reply:
The title could be a ReactNode as well. There might be times, though not often, where the tooltip has some markup within it.